vertx-core/src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -996,4 +1064,36 @@ public static String positiveLongToString(long value) {
     }
     return str;
   }
+
+  public static boolean isFrameBased(io.vertx.core.http.HttpVersion version) {
+    return FRAME_BASED_VERSIONS.contains(version);
+  }
+
+  public static boolean isHttp3(io.vertx.core.http.HttpVersion protocolVersion) {
+    return HTTP3_VERSIONS.contains(protocolVersion);
+  }
+
+  public static boolean supportsQuic(List<String>applicationProtocols) {
+    if (applicationProtocols == null || applicationProtocols.isEmpty()) {
```

Comment:
we should not check against `applicationProtocols` being null, if that happens then the caller should not call this method otherwise it is a bug